google-cloud-storage/src/main/scala/akka/stream/alpakka/googlecloud/storage/StorageObject.scala

```diff
@@ -140,7 +190,13 @@ object StorageObject {
       etag: String,
       md5Hash: String,
       crc32c: String,
-      mediaLink: String
+      mediaLink: String,
```

Comment:
Modifying `StorageObject.apply` and `StorageObject.create` will cause a binary incompatibility with this API.  Ideally we would like to avoid this scenario because a user may have code that already exists to create `StorageObjects` to transmit to the google-cloud-storage connector's `Sink` or `Flow` APIs.  If there's a good reason to modify these methods (and the `with` builder methods are not good enough), then we should consider adding constructor overloads to accomodate the extra parameters.  Do you need to modify these methods?

Reply:
Hi @seglo, good point! I believe the `with` builder methods should be good enough, I'll amend my PR.

Reply:
Thanks.  You'll also be able to revert these two MiMa exceptions, but keep the one on `this`.

```
ProblemFilters.exclude[DirectMissingMethodProblem]("akka.stream.alpakka.googlecloud.storage.StorageObject.apply")
ProblemFilters.exclude[DirectMissingMethodProblem]("akka.stream.alpakka.googlecloud.storage.StorageObject.create")
```

Reply:
hi @seglo, just before I go off and do this, the only way I can see to implement this is by making the parameters I'm adding optional and giving them a default value of none. 
Is that ok?

Reply:
Hi @DanieleSassoli.  Thanks for checking in, it made me take a closer look at this.  I looked at the [Google Cloud Storage API spec](https://cloud.google.com/storage/docs/json_api/v1/objects#resource-representations) and noticed the additional fields you would like to add.  I assume that when this Alpakka connector was first created, that these fields didn't exist and that they were added later on as non-breaking changes by Google (non-breaking because it's just a JSON API).

I also noticed that nowhere in the Alpakka DSL do we actually expose a way for users to create their own `StorageObject` for the purpose of writing data to google cloud.  We only let the user provide a storage object name identifier and a `ByteString`, and `StorageObjects` are only ever deserialized from google API responses.  In fact, I don't see any internal use of our `with` methods either, so I assume these were added at some point for reasons that are unclear to me (maybe for internal testing?)

Therefore I don't see any reason for not modifying the constructor with additional fields as your PR originally intended.  
Since you are more familiar with this API I would like your advice on how to proceed.  Is my understanding of the usage of the `StorageObject` correct?  Is there ever a reason (using this Alpakka connector) to mutate fields of the `StorageObject`?  I notice that the spec suggests some fields are "writable", but I don't think that's relevant to this Alpakka connector?

Reply:
Thanks for looking into this and getting back to me. 
Your analysis makes sense, the only reason we have to generate a `StorageObject` and mutate its fields is for testing purposes. 
I've been able to generate an `Arbitrary[StorageObject]` via the `StorageObject.apply` method, this allows me to generate random instances of `StorageObject` for my tests, I then use the `with` method to modify those 2 or 3 fields that I need to have control over(like `name` and `bucket`).

You're also correct when you say that within the API they're only ever deserialised from responses. 

Does this answer your questions?

Reply:
No, not all the fields are there at the moment, I think a few are still missing as I didn't know if this was of any interest so to begin with I only took the ones I considered more relevant. 
I've got a busy day today and tomorrow but will try to get this sorted before the weekend. 
Thanks again for your help.